go.mod

```diff
@@ -1,51 +1,68 @@
 module github.com/yarpc/yab
 
-go 1.14
+go 1.18
```

Comment:
why are we doing this?

I thought you previously recommended we don't update unless necessary? 

I was reading about the unused module pruning in go 1.17 recently though - I was wondering if that means we can drop the `tools` sub-package.

Reply:
I might have been misinformed when I advised against bumping this.

I thought that for libraries, the `go` directive specifies the minimum required Go version but that's not true. A library with a 1.18 directive will still build on 1.17 as long as 1.18-based features are gated behind build tags.

For binaries, it matters even less.
So it makes sense to make use of the graph pruning introduced in 1.17.
(go.sum tries to remain compatible with N-1 version, so graph pruning doesn't clean it up until 1.18.)